### PR TITLE
ipkg-build fails when called with a relative path

### PR DESCRIPTION
**Pull request by KOLANICH**

### CODE REVIEW

**Comment by KOLANICH** [comment]
Also I wanna note the following:
* that I used this script as a workaround to `opkg` bug that `opkg` refuses to install a package with incorrect deps (in particular, kernel commit hash, I had to reforge some kernel modules packages in order to make them installable with opkg onto a kernel with a different hash (but is close in terms of commits, so must be compatible) ).
* it may make sense to rewrite this script in python for better integration with other code written in python. Do you think it makes sense to replace it here?

**Comment by adschm** [comment]
Hi, please note that a full proper name in Signed-off-by is mandatory. Will the e-mail address work, since it has "noreply" in it?

**Comment by KOLANICH** [comment]
>Hi, please note that a full proper name in Signed-off-by is mandatory.

It is not: you currently have no means to detect fake names looking like real persons' names (and that could really be persons' names) from real ones. If you really wanted and needed people signing commits using their real legal names you would have to do the procedures like in banks. If you really want be more sure, you can also plug some biometric system here, like one used in EU and US. If you are not ready (expect a drop in contributions if you would really do that, in fact projects for which geting guaranteedly copyrightly-clean code is really crucial (lke SQLite, havinghighly paying clients are aerospace and automative) just don't accept contributions at all) and/or cannot do that, it makes sense to stop this hypocrisy.

>Will the e-mail address work, since it has "noreply" in it?

Of course no, it is a dummy "email address" used by GitHub to link commits to a GitHub account (since git requires a email address to be set).

File: scripts/ipkg-build [hunk 1]
```diff
@@ -125,7 +126,7 @@ case $# in
 	;;
 esac
 
-pkg_dir=$1
+pkg_dir=`$REALPATH $1`
```
Comment (aparcar) [inline scripts/ipkg-build:129]: Please use $() and wrap in in `"`.

**Review by aparcar — CHANGES_REQUESTED** [review]

**Comment by aparcar** [comment]
> It is not: you currently have no means to detect fake names looking like real persons' names (and that could really be persons' names) from real ones. If you really wanted and needed people signing commits using their real legal names you would have to do the procedures like in banks. If you really want be more sure, you can also plug some biometric system here, like one used in EU and US. If you are not ready (expect a drop in contributions if you would really do that, in fact projects for which geting guaranteedly copyrightly-clean code is really crucial (lke SQLite, havinghighly paying clients are aerospace and automative) just don't accept contributions at all) and/or cannot do that, it makes sense to stop this hypocrisy.

OpenWrt contains a lot of Kernel patches which ideally end up in future Kernel releases rather than being maintained by OpenWrt itself indefinitely. The Kernel requires *real names* for the reasons described [here](https://www.kernel.org/doc/html/v5.10/process/submitting-patches.html#developer-s-certificate-of-origin-1-1). To allow OpenWrt patches to become part of the Kernel, we adopted this requirement.

There is no (reasonable) way to verify this origin, however an obvious nickname is simply not accepted.

> Of course no, it is a dummy "email address" used by GitHub to link commits to a GitHub account (since git requires a email address to be set).

Real mail addresses are helpful to follow up with developers in the future.

If you require anonymity *and* this feature I can fix multiple shellcheck errors in `ipkg-build` and additionally add `realpath` in a similar way.

**Comment by adschm** [comment]
Since the author clearly expressed that he does not intend to comply to our guidelines, this is closed.

**Comment by KOLANICH** [comment]
@aparcar,

>OpenWrt contains a lot of Kernel patches which ideally end up in future Kernel releases rather than being maintained by OpenWrt itself indefinitely.
>To allow OpenWrt patches to become part of the Kernel, we adopted this requirement.

Large part of OpenWRT is not kernel patches. And this PR doesn't touch any kernel code.

>The Kernel requires real names for the reasons described here.

The provided link address this with the following remark:

>using your real name (sorry, no pseudonyms or anonymous contributions.)

Looks like just an extremily reasonable reason !

>Real mail addresses are helpful to follow up with developers in the future.

My real email address is not a big secret, one can find it with some effort. But inserting it in cleartext anywhere is not a very good idea.

>If you require anonymity

Pseudonimity and anonymity are different things. Anonimity means no attribution at all.

>and this feature I can fix multiple shellcheck errors in ipkg-build and additionally add realpath in a similar way.

In fact I am going (but no ETA) to rewrite this script into python 3 entirely and maybe make a package. It would allow to use it as a lib. But the fact that the script cannot be called using relative paths is clearly a bug that should be fixed. Feel free to take this PR, address the review comments, put your name to it and send it to this repo. All my original contributions, contained in this patch/PR, can be considered as licensed under [Unlicense](https://unlicense.org/), so everyone in the world is allowed to take it and put an own name on it.

**Comment by aparcar** [comment]
> In fact I am going (but no ETA) to rewrite this script into python 3

Please be aware that OPKG *may* be replaced by APK in some future. The current work happened [here](https://github.com/openwrt/openwrt/pull/3811).

**Comment by KOLANICH** [comment]
Thanks for the info!

**Pull request closed without merging**